### PR TITLE
Add mentor titles + display mentor descriptions for lessons

### PR DESCRIPTION
**Pull request by callumacrae**
## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDEL-472

We want to xxx because yyy.

### Is there anything weird that code reviewer should know?
Nope, everything is straightforward.

### Review Checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests
- [x] All forms have an `id` attribute on them

### How can someone see it it review app?

1. Click the link to review app (posted by a `github-actions` bot below)
2. Do something

### CODE REVIEW

File: app/views/core_induction_programmes/modules/show.html.erb [hunk 1]
```diff
@@ -26,12 +26,26 @@
...
+            <% end %>
+            <% if lesson.mentor_summary %>
+              <h3 class="govuk-heading-s">Mentor summary:</h3>
+              <p class="govuk-body"><%= lesson.mentor_summary.presence || "No summary available" %></p>
+            <% end %>
+          <% elsif current_user.mentor? %>
+            <p class="govuk-body"><%= lesson.mentor_summary.presence || "No summary available" %></p>
+          <% else %>
+            <p class="govuk-body"><%= lesson.ect_summary.presence || "No summary available" %></p>
+          <% end %>
```
Comment (kicferk1) [inline app/views/core_induction_programmes/modules/show.html.erb:48]: Hmm, I think summaries should behave like titles - when there is no mentor summary, always show ect summary. (we do have ect summaries for modules everywhere, but no mentor summaries for them, and, ehh, who knows how soon we will get that sorted).

Comment (kicferk1) [reply app/views/core_induction_programmes/modules/show.html.erb:48]: Feel free to make a separate PR for it if you want :)

**Comment by kicferk1** [comment]
I think cypress tests are failing because the step definition for ect title is not specific enough now.

`  "title input": '[name*="title"]',` is what's catching two inputs and breaking cypress :(

File: app/views/core_induction_programmes/modules/show.html.erb (hunk 1 above)
Comment (callumacrae) [reply app/views/core_induction_programmes/modules/show.html.erb:48]: I'm using the behaviour specified on the ticket, which looked pretty specific: https://dfedigital.atlassian.net/browse/CPDEL-472

Comment (kicferk1) [reply app/views/core_induction_programmes/modules/show.html.erb:48]: > GIVEN the user is a mentor
THEN display the mentor-specific weekly description on the weekly breakdown screen

Nothing about having a `No summary available`. If anything, we should display absolutely nothing if the summary is missing. Could you double check with Kam?

Comment (callumacrae) [reply app/views/core_induction_programmes/modules/show.html.erb:48]: kam says leave blank! i'll change it

File: spec/cypress/integration/CipLessons.feature [hunk 2]
```diff
@@ -11,8 +11,8 @@ Feature: Core Induction Programme lessons
     And the page should be accessible
     And percy should be sent snapshot called "edit lesson page"
 
-    When I clear "title input"
-    And I type "New lesson title" into "title input"
+    When I clear "year title input"
+    And I type "New lesson title" into "year title input"
```
[the remark this answers is not in the document]
Comment (callumacrae) [reply spec/cypress/integration/CipLessons.feature:15]: …good point 🤦

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**